src/lib/Monads.ml

```diff
@@ -154,15 +155,14 @@ struct
       abort_if_inconsistent () @@
       match cofs with
       | [] -> m
-      | (Cof.Var _ | Cof.Cof (Cof.Top | Cof.Bot | Cof.Eq _)) as phi :: cofs ->
+      | (Cof.Var _ | Cof.Cof (Cof.Meet [] | Cof.Join [] | Cof.Eq _)) as phi :: cofs ->
```

Comment:
Question: do we need special cases for `Cof.Meet []` and `Cof.Join []` here?

Reply:
Yes.

Reply:
Are you sure? `restrict (Cof.Meet [])` seems to be doing nothing. Not sure about `restrict (Cof.Join [])`, though.